packages/keyring-controller/src/KeyringController.ts

```diff
@@ -2091,13 +2109,83 @@ export class KeyringController extends BaseController<
       delete state.encryptionSalt;
     });
 
-    this.#password = password;
+    await this.#deriveEncryptionKey(password);
 
     await this.#clearKeyrings();
     await this.#createKeyringWithFirstAccount(keyring.type, keyring.opts);
     this.#setUnlocked();
   }
 
+  /**
+   * Derive the vault encryption key from the provided password, and
+   * assign it to the class variable for later use with cryptographic
```

Comment:
Is it a class variable? I might be confused with the terminology, but I think class variables are `static`.